kyo-prelude/shared/src/main/scala/kyo2/kernel/Effect.scala

```diff
@@ -209,10 +209,11 @@ object Effect:
                         val tag   = kyo.tag
                         val input = kyo.input
                         def frame = _frame
-                        def apply(v: OX[Any], context: Context)(using Safepoint) =
+                        def apply(v: OX[Any], context: Context)(using safepoint: Safepoint) =
                             try catchingLoop(kyo(v, context))
                             catch
                                 case ex: Throwable if (NonFatal(ex) && pf.isDefinedAt(ex)) =>
+                                    Safepoint.insertTrace(ex)
```

Comment:
It isn't enough to handle exceptions only in `Safepoint.eval` since exceptions can be caught before reaching `eval`. All APIs that provide exception handling have to use `Effect.catching` (including `Abort`) so we can introduce the trace here as well.

Reply:
Doesn't it make more sense to capture the trace only if the partial function is not defined?

Reply:
When the partial function is defined, the exception "escapes" the scope of the kernel evaluation (typically into a `Result` via `Abort`) so it's essentially the last opportunity to inject the trace. In case the partial function is undefined, the exception still gets the trace included either by an outer `Effect.catching` call or `<.eval`.

Reply:
Ah, that makes more sense.